extension/httpforwarderextension/extension.go

```diff
@@ -60,7 +64,9 @@ func (h *httpForwarder) Shutdown(_ context.Context) error {
 	if h.server == nil {
 		return nil
 	}
-	return h.server.Close()
+	err := h.server.Close()
+	h.shutdownWG.Wait()
+	return err
```

Comment:
Why not using [Shutdown](https://cs.opensource.google/go/go/+/go1.23.6:src/net/http/server.go;l=3049) [¶](https://pkg.go.dev/net/http#Server.Shutdown)?

Reply:
The use of `Shutdown` vs `Close` is orthogonal to the issue being fixed in this PR: deterministic release of the port used by the component. In both cases there is still the need to wait the call to `Serve(listener)` to return.

I tend to prefer `Close` since, in certain cases `Shutdown` can take much longer since it needs to wait existing connections to finish gracefully. That said the component owners will have more context if they want to change the current behavior in that regard.

Reply:
I may miss something:

> // Shutdown gracefully shuts down the server without interrupting any
// active connections. Shutdown works by first closing all open
// listeners, then closing all idle connections, and then waiting
// indefinitely for connections to return to idle and then shut down.

Reply:
Take into account the possible interleaves of the calls to `Serve` and `Shutdown`/`Close`. The ownership of `listener` is transferred to the `http.Server` when the call to `Serve(listener)` happens. However, since that happens in a goroutine it is possible that `Shutdown` is called before the `http.Server` took the ownership of listener. In this case the listener is not closed until the call to `Serve` happens. `Serve` adds a deferred call to `listener.Close`, even if `Shutdown` was already called, which ensures that the port is released by the time the method returns. 

While this may be rather improbable in practice, it is better for the component to behave deterministically with the shutdown ensuring no left over goroutine and/or port.